concreep-refilled/control.lua

```diff
@@ -144,7 +151,11 @@ function _reinit_reentrant_level1_start()
 	global.no_work_counter = 0
 
 	-- set up outer arrays to iterate
-	global.reinit_surfaces = array_shallowcopy(game.surfaces)
+	for str in SURFACES:gmatch("([^,]+)") do
+	name = str:gsub("%s+", "")
+    local surf = game.surfaces[name]
+	end
+	global.reinit_surfaces = array_shallowcopy(surf)

```

Comment:
> A bit hesitant here - first I'm struggling a bit iterate through the surfaces - for some reason i mess that up every time I try. Secondly, isn't it cheaper to never iterate through all the game surfaces but only through the ones we specify?

the reason to iterate and filter later is in case a surface gets created after initialization of our settings

> Also we should add a check for when no valid surfaces are supplied to just use nauvis -> or print a message that Concreep is disabled till a valid surface is entered

hmmm yea, probably print a message in ... init if the trimmed version of surfaces allowed is empty string